odxtools/singleecujob.py

```diff
@@ -183,7 +189,14 @@ def from_et(et_element: Element, doc_frags: List[OdxDocFragment]) -> "ProgCode":
 
         encryption = et_element.findtext("ENCRYPTION")
 
-        syntax = et_element.findtext("SYNTAX")
+        syntax = ProgCodeSyntax("JAR")
+        try:
+            syntax = ProgCodeSyntax(odxrequire(et_element.findtext("SYNTAX")))
+        except ValueError as e:
+            try:
+                odxraise(f"Could not parse program code syntax")
+            except Exception as ee:
+                raise ee from e
```

Comment:
what's going on here?

Reply:
we want to raise an `OdxException` instead of `ValueError` while keeping the traceback of the original exception, but we only want to throw anything if we are in strict mode. Maybe odxraise could be extended so that an alternative traceback can be specified, but that would probably not really make things more intuitive?

Reply:
a simple `if value in ProgCodeSyntax.__members__:` would have been enough

Reply:
good point. fixed!